### PR TITLE
doc: Add ShellCheck to lint tests dependencies

### PR DESCRIPTION
**Pull request by hebasto**
In master (9641366950276da88af626d0898676195df8d83a) the lint tests dependencies list lacks ShellCheck. This PR fixes it.

Also `lint-python.sh` is slightly improved.

### CODE REVIEW

**Comment by practicalswift** [comment]
ACK 0f7051a02091bd87d2502b8249f5ad977f3a0b84 -- diff looks correct

**Review by promag — COMMENTED** [review]
ACK 0f7051a02091bd87d2502b8249f5ad977f3a0b84.

File: test/lint/lint-python.sh [hunk 1]
```diff
@@ -81,14 +89,6 @@ enabled=(
     W606 # 'async' and 'await' are reserved keywords starting with Python 3.7
 )
 
-if ! command -v flake8 > /dev/null; then
```
Comment (laanwj) [inline test/lint/lint-python.sh]: Does this cause any problems?

I think I prefer leaving it here; to have the list of exceptions (a "constant") above the code as much as possible.

Comment (promag) [reply test/lint/lint-python.sh]: LGTM.

Comment (hebasto) [reply test/lint/lint-python.sh]: > I think I prefer leaving it here; to have the list of exceptions (a "constant") above the code as much as possible.

Agree. Fixed.

**Comment by laanwj** [comment]
ACK 80c9e66ab84f8cecc2bf2eebf508a5aad8911246

Thanks for improving documentation on running these locally. I hope it helps avoid frustrations like #17346.

**Comment by promag** [comment]
ACK 80c9e66ab84f8cecc2bf2eebf508a5aad8911246, verified internal and external links. Nice looking table.

**Pull request merged**